src/accounts_refactor.rs

```diff
@@ -160,62 +138,58 @@ impl<'a, L: Ledger, Key: KeyPair + DeserializeOwned + Serialize> AccountEditor<'
     /// Set the account as used.
     pub fn set_used(mut self) -> Self {
         self.account.used = true;
-        self.account.modified_time = Local::now();
-        self
-    }
-
-    /// Set the account as unused.
-    pub fn set_unused(mut self) -> Self {
-        self.account.used = false;
-        self.account.modified_time = Local::now();
         self
     }
 
     /// Set the optional ledger scan.
-    pub fn set_scan(mut self, scan: Option<BackgroundKeyScan<L>>) -> Self {
+    pub(crate) fn set_scan(mut self, scan: Option<BackgroundKeyScan<L>>) -> Self {
         self.account.scan = scan;
-        self.account.modified_time = Local::now();
         self
     }
 
     /// Set the ledger scan.
-    pub fn with_scan(mut self, scan: BackgroundKeyScan<L>) -> Self {
+    pub(crate) fn with_scan(mut self, scan: BackgroundKeyScan<L>) -> Self {
         self.account.scan = Some(scan);
-        self.account.modified_time = Local::now();
         self
     }
 
     /// Clear the leger scan.
-    pub fn clear_scan(mut self) -> Self {
+    pub(crate) fn clear_scan(mut self) -> Self {
         self.account.scan = None;
-        self.account.modified_time = Local::now();
         self
     }
 
     /// Update the ledger scan.
+    ///
+    /// Returns
+    /// * `Err` if the scan isn't found, or
+    /// * `Ok((self, scan_status))`, where `scan_status` provides the scanned information if the
+    /// scan is complete, or the scan itself if it's in progress.
     pub(crate) async fn update_scan(
         mut self,
         event: LedgerEvent<L>,
         source: EventSource,
         records_commitment: MerkleCommitment,
-    ) -> Result<AccountEditor<'a, L, Key>, KeystoreError<L>> {
+    ) -> Result<(AccountEditor<'a, L, Key>, ScanStatus<L>), KeystoreError<L>> {
```

Comment:
Might be better to just return `Result<Option<ScanOutputs>>` here, so we don't have to clone the scan when we update `self.account.scan` in the `InProgress` case. Don't think it's a big deal though